multitenancy/tenant-manager-client/src/main/java/io/apicurio/multitenant/client/TenantManagerClientImpl.java

```diff
@@ -15,70 +15,71 @@
 import io.apicurio.multitenant.api.datamodel.SortBy;
 import io.apicurio.multitenant.api.datamodel.SortOrder;
 import io.apicurio.multitenant.api.datamodel.TenantStatusValue;
 import io.apicurio.multitenant.api.datamodel.UpdateRegistryTenantRequest;
+import io.apicurio.multitenant.client.exception.TenantManagerClientErrorHandler;
+import io.apicurio.multitenant.client.exception.TenantManagerClientException;
+import io.apicurio.rest.client.JdkHttpClient;
+import io.apicurio.rest.client.VertxHttpClient;
+import io.apicurio.rest.client.auth.Auth;
+import io.apicurio.rest.client.auth.exception.AuthErrorHandler;
+import io.apicurio.rest.client.request.Request;
+import io.apicurio.rest.client.spi.ApicurioHttpClient;
+import io.apicurio.rest.client.util.IoUtil;
+import io.vertx.core.Vertx;
 
-import org.apache.http.NameValuePair;
-import org.apache.http.client.utils.URIBuilder;
-import org.apache.http.message.BasicNameValuePair;
-import org.keycloak.common.VerificationException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.apicurio.rest.client.request.Operation.DELETE;
+import static io.apicurio.rest.client.request.Operation.GET;
+import static io.apicurio.rest.client.request.Operation.POST;
+import static io.apicurio.rest.client.request.Operation.PUT;
 
 /**
  * @author Fabian Martinez
  */
 public class TenantManagerClientImpl implements TenantManagerClient {
 
     private static final String TENANTS_API_BASE_PATH = "api/v1/tenants";
+    private static final String TENANTS_API_BASE_PATH_TENANT_PARAM = "api/v1/tenants/%s";
 
-    private final HttpClient client;
+    private final ApicurioHttpClient client;
     private final ObjectMapper mapper;
-    private final String endpoint;
-    private final Auth auth;
 
     public TenantManagerClientImpl(String endpoint) {
-        this(endpoint, null);
-
+        this(endpoint, Collections.emptyMap(), null);
     }
 
-    public TenantManagerClientImpl(String endpoint, Auth auth) {
-        if (!endpoint.endsWith("/")) {
-            endpoint += "/";
+    public TenantManagerClientImpl(String baseUrl, Map<String, Object> configs, Auth auth) {
+        if (!baseUrl.endsWith("/")) {
+            baseUrl += "/";
         }
-        this.endpoint = endpoint;
-        this.client = HttpClient.newHttpClient();
+        final String endpoint = baseUrl;
         this.mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-        this.auth = auth;
+        this.client = new JdkHttpClient(endpoint, configs, auth, new TenantManagerClientErrorHandler());
+    }
+
+    public TenantManagerClientImpl(Vertx vertx, String baseUrl, Map<String, Object> configs, Auth auth) {
+        if (!baseUrl.endsWith("/")) {
+            baseUrl += "/";
+        }
+        final String endpoint = baseUrl;
+        this.mapper = new ObjectMapper();
```

Comment:
this should be `this.mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);`

Reply:
Oh yes, sorry, this is probably a rebase issue.